tasks/sync_pull.py

```diff
@@ -379,9 +393,30 @@ def update_pull_commits(
             )
         return {"soft_deleted_count": deleted_count, "merged_count": merged_count}
 
-    def was_pr_merged_with_squash(
-        self, commits_on_pr: Sequence[str], base_ancestors_tree: Dict[str, Any]
-    ) -> bool:
+    def was_squash_via_merge_commit(
+        self, repoid, pullid, repository_service, pull_dict
+    ):
+        # if the merge commit exists for this PR, and that commit
+        # has multiple parents, then it's a regular merge commit
+        # otherwise it's a squash
+
+        merge_commit_sha = pull_dict.get("merge_commit_sha")
+        log.info(
+            "Sync Pull using merge commit sha experiment running",
+            extra=dict(repoid=repoid, pullid=pullid, merge_commit_sha=merge_commit_sha),
+        )
+        if merge_commit_sha is not None:
```

Comment:
```suggestion
        if merge_commit_sha is None:
            return None
```

Just a personal preference: early-return, and then un-indent the normal code path